### PR TITLE
Use the board button to Reset the board and also to get into boot-loader mode

### PR DESCRIPTION
**Pull request by nrjn**
Press the board button for more than 1 second to reset the board.
Press and release the button immediately to get into the boot-loader mode.

### CODE REVIEW

File: board/ubirch1r03/board.c [hunk 1]
```diff
@@ -45,3 +46,62 @@ void board_install_bootloader_hook(void) {
...
   runBootloader(NULL);
 }
+
+volatile uint32_t lptmrCounter = 0U;
+volatile bool button_press_status = false;
+
+void BOARD_LPTMR_HANDLER(void)
+{
+  LPTMR_ClearStatusFlags(LPTMR0, kLPTMR_TimerCompareFlag);
+  lptmrCounter++;
```
Comment (thinkberg) [inline board/ubirch1r03/board.c]: check if you really need this counter, if you set the lptimer to run for exactly one second, you can just check if it stopped.

File: board/ubirch1r03/board.c [hunk 2]
```diff
@@ -45,3 +46,62 @@ void board_install_bootloader_hook(void) {
...
+  /*
+   * Workaround for TWR-KV58: because write buffer is enabled, adding
+   * memory barrier instructions to make sure clearing interrupt flag completed
+   * before go out ISR
+   */
+  __DSB();
+  __ISB();
+}
+
+int Board_Reset_Bootloader(void) {
```
Comment (thinkberg) [inline board/ubirch1r03/board.c]: lower case function names please

File: board/ubirch1r03/board.h [hunk 3]
```diff
@@ -87,15 +94,23 @@ static inline void board_init() {
...
 
   PORT_SetPinMux(BOARD_PWR_EN_PORT, BOARD_PWR_EN_PIN, kPORT_MuxAsGpio);
   GPIO_PinInit(BOARD_PWR_EN_GPIO, BOARD_PWR_EN_PIN, &OUTFALSE);
 
   // installs a bootloader hook
+
   board_install_bootloader_hook();
 
   // enable NMI handler, use it to call the bootloader
-  SCB->SHCSR = SCB_ICSR_NMIPENDSET_Msk;
```
Comment (thinkberg) [inline board/ubirch1r03/board.h]: completely remove it if not used

File: board/ubirch1r03/board.c (hunk 1 above)
Comment (niranjanraoh) [inline board/ubirch1r03/board.c]: removed the counter

**Pull request merged**